autoload/neomake/makers/ft/rust.vim

```diff
@@ -136,15 +137,18 @@ function! neomake#makers#ft#rust#CargoProcessOutput(context) abort
 
         let error = {'maker_name': 'cargo'}
         let code_dict = get(data, 'code', -1)
-        if code_dict is g:neomake#compat#json_null
-            if get(data, 'level', '') ==# 'warning'
-                let error.type = 'W'
-            else
-                let error.type = 'E'
-            endif
+        if get(data, 'level', '') ==# 'warning'
+            let error.type = 'W'
         else
-            let error.type = code_dict['code'][0]
-            let error.nr = code_dict['code'][1:]
+            let error.type = 'E'
+        endif
+        if code_dict isnot g:neomake#compat#json_null
+            if code_dict['code'][0] ==# 'W' || code_dict['code'][0] ==# 'E'
```

Comment:
`code_dict` might be -1 here, from the default above?!

Reply:
Maybe `g:neomake#compat#json_null` should be used as default above? (with `get(data, 'code', …)`)

Reply:
fixed